src/main.js

```diff
@@ -1827,12 +1841,20 @@ const DiscoveryFeedApplication = new Lang.Class({
         if (this._debugWindow)
             return;
 
-        let gdkWindow = this._window.get_window();
-        gdkWindow.set_events(gdkWindow.get_events() |
-                             Gdk.EventMask.FOCUS_CHANGE_MASK);
-        this._window.connect('focus-out-event', Lang.bind(this, function() {
-            this.hide();
-            return false;
+        // Connecting to 'draw' here is not ideal, however it seems like
+        // the 'realize' signal is not fired for GtkWindow when it gets
+        // mapped
```

Comment:
Pretty sure this is because we already call `realize()` on a children before calling into this: https://github.com/endlessm/eos-discovery-feed/pull/73/commits/8b25b4dc7038f91ef02359457237412c004f1dc4#diff-1c90ff38a08209f9ebd4d05d1e43358eR1726

Reply:
Ah interesting. Should we connect to the `realize` signal and make that call below instead then?

Reply:
I actually don't think there's any change needed in how we handle the connection to `focus-out-event`. Probably this was required when you first wrote the PR, and we still depended on Wnck, but not right now. I'd just drop that commit.

Reply:
(For now I've rebased to connect to `realize` before calling `realize` on the child widget).

I think I see what is going on here. The relevant change is not from when we dropped Wnck, but from when the child content of the window was explicitly realized, causing the the rest of the window to be realized too. That meant that calling `gtk_window_get_window` would have worked, when it would not have worked before and we would have had to wait for the `realize` signal (or first draw signal) before we did that.

I think we still want this commit, or at least a version of it. Depending on the underlying window being realized at this point because we happened to explicitly realize a child widget seems wrong to me - the idiomatic way has always been to explicitly wait for the `realize` signal before doing anything that requires the underlying `GdkWindow`.

I'll change the commit message to make this more clear.

Reply:
Also confusing matters is that we keep track of the `focus-out-event` signal ID in this commit too. I'll separate that out.